### DISCUSSION TITLE
The Mega Bug/Exploit List (by StarDust Fox)

### DISCUSSION THREAD

**Discussion by EmilitiaEnnehrt** [Equinox-SS13/equinox-sojourn#36] [General]
### Foreword from Emilitia.
> The following is the requested (by me) bug/exploit list that has been known over the years by a particular regular player from the original live server that has been documenting and reporting bug fixes for a long time.

> In this case, I'll be taking over maintenance operations of this, striking off the list as we go over and fix or verify them as being fixed or cannot be replicated. I've also changed the list around and categorize them for better priorty/viewing.

> Do know that some of these may be old bug or already fixed. Talk to the person to inquire more. Additionally, "circuit bugs are a beast by itself" - stardust.

> None of the content after this line is written by me, so all further questions and inquiry should go directly to the member; `stardustfox`
---

The bugs and exploits present here are a collection of me doing, seeing and stumbling upon, any doubt regarding anything in this list and other doubts of similar nature, refer to stardustfox. in discord (StarDust Fox.)

### General Exploits

- [x] ~~Food and insight: Food, the one which the chef can produce or anyone with access to a microwave (or 10 steel and 1 wood plank to make a burning barrel), can be eaten to gain sanity back and majority of cases also insight to satisfy your need, you can inject water in said food to make it constantly give insight and sanity back, making one level up absurdly quick (The fastest I reached 150 in all stats was 40 minutes in round from the start, no core implants)~~ #59 

- [ ] Fireman throwing through windows: Doing a fireman throw forceMoves the mob onto the turf behind you even if that turf is a wall or window

- [x] ~~Cleansing Carbine plasteel exploit: Bioprinter can make cleansing carbines using 4.5 steel, you can use a wire cutter to dismantle said carbines into 4 steel and a misc gun part, said misc gun part when recycled yields 5 plasteel, 0.5 steel to 5 plasteel alchemy.~~ #49

- [x]  ~~Church armor bundle exploit: The church armor bundles cost 15 biomatter, but upon being selected, you can recycle the two pieces of gear for a return of 30 biomatter, resulting in more biomatter than originally spent.~~ #49

- [ ] Prime’s Sword: The Joyeuse's special ability to cause a flash when used in both hands can be spammed via hotkey and clicking to chain react a big wave of LOUD blinding flashbangs (Which can be used against pretty much anyone that doesn't bear a cruciform AND blobs to insta kill all their nodes in view)

- [x] ~~Church altars: They currently have a system to stop one from calling cruciforms or church cybernetics each 10 minutes, but one can still spam the litany to bring in several at once.~~ #49 

### Church

- [ ] Church golems: They appear to take damage from biomass, even if slightly, at least enough to cause their examine text to say they did get damage, and be able to be repaired from such using welders.
 
- [x] ~~Church concilium LMG: It has several firemodes, and a normal fire rate, but upon selecting a different fire mode, the gun’s fire rate breaks, making it slower than before, this causes the effect of the gun losing its speed but also becoming immune to any fire rate changing mod, either to speed it up or slow it down.~~ #49 
 
- [ ] Church Martyr Pistol: Upon firing the whole gun and spending it empty, one can click it in hand to remove its cell (robust cell 100), unable to place back in, still odd considering the gun is only in desc said to be Biomatter to plasma conversion.
 
- [ ]  Church armor inconsistency: Rosaria helmet has the same melee protection (7) as the basic vector helmet, while the angel’s suit helmet has better melee protection (10) considering its cheaper and doesn’t cost as much plasteel as the Rosaria (30 plasteel).

- [ ] Church’s ashen stone: Originally it was meant to be placed right between the 4 rocks it is surrounded by.

#### RIG Suit Bugs
- [ ] Rig Suit and modules: Soteria can produce several modules, but their prices are strangely high if compared to the price they are on Lonestar’s beacon, making it far more advantageous to buy the modules rather than getting it printed by SI, a few of those also requiring unusual materials, like the night vision rig module requiring platinum instead of uranium, making it more cost (and use) effective to get someone to surgically install NVGs cybernetics.

- [ ] Some modules are also broken, for example night vision mod which stops working in case the operator is also using mesons.

- [ ] Using a tape on the rig gauntlets will stick the gauntlets to your in-hand, making you unable to use items, as well deleting the items in your pockets.

- [ ] Modules still work even if they are said to be broken via the hardsuit screen

- [ ] Night vision doesn't work together with mesons, breaking night vision, but mesons still work (Both rig modules)

- [ ] Rig suits security HUD doesn't work

- [ ] Rig suit energy blade has a redundant select button

- [ ] The RIG net projector has also never worked as long as I can remember here. Same issue (I believe) effects the suspension field generator mechanic for grabbing people
---

- [ ] Glass panel and mob A.I: If a window is placed sideways through a narrow spot, it will block A.I from being able to both damage said window nor pass (As image shows)
![image](https://github.com/user-attachments/assets/4a476ddb-2e7f-4d2f-9cfe-aba9b4e56406)

- [ ] Advanced syrette: They used to have a different sprite before and actually show their liquid content, now they are just gray basic looking normal syrettes impossible to change color.

- [ ] Paint can’t get cleaned from FBPs (and likely others?): If you splash yourself with paint (either crayon dust or actual paint), you are unable to get rid of that, tried shower, space cleaner, rejuv agent, walking inside pool, removing all clothing and showering or spraying again, as an FBP at least nothing would clear the painting, only “fix” was to paint myself white.

- [ ] Cht surgical gland (and likely others): Cht's, more specifically the Ru's surgical gland can be inserted into someone via surgery, such can be caused by accident and thus breaks the Cht's hand making it unable to pick anything else, the same can happen with other cybernetic surgical tools or basic toolset in case of synthetics

- [ ] Synthetic organs need prying: prosthetic organs require a tool with prying, like a crowbar, but clicking on the button in hand with it doesn't do anything.

- [ ] Mycus and folken getting infections: Something that seems to be recurring, Mycus normally shouldn’t get any infections, but in some rare strange cases they seem to still acquire infections, similarly to Folken.

- [ ] Mob and fire sprite: As of now sometimes mobs can catch on fire, emit the red fiery light but not show their sprites as being on fire.

- [ ] Pool and cleaning solution: Pool tiles get "cleaned" if hit by cleaning solution to be not blue. 
> Emi's addition.. "wooden" doors too.

- [ ] Disk Storages: They can’t be dismantled nor wrenched to be moved.

- [ ] Material box breaks sheet sprites: Easy fix because one just needs to pick them up to fix their sprites, but when I emptied the box of materials, they turned invisible and could only be seen by using the turf menu. Picking them up fixed them.

- [ ] Examining trash piles: Upon examining a trash pile, it updates all its contents, from that, two interactions can happen: Either you get shocked by a malfunctioning rig that is inside that pile, or rarely a faulty atom cell will do its recharge tick and cause the pile to explode.

- [ ] Welding tool: You can start dismantling a wall with the welder, then turn it off, it will stop using its fuel and still dismantle said wall.
---

### All About Hydrogen (and Exploits)

"Classia" hydrogen-plasma pistol (Heat generation of 25)
"Venatori" hydrogen-plasma gun (Heat generation of 25)
"Sollex" hydrogen-plasma cannon (Heat generation of 50)

Reading on code they have a heat generation, basically 4 shots of Classia, Venatori are safe (25), in quick rush, the 5th shot will cause the malus effect Sollex has 50, 2 shots are max safety, 3th is the malus. 

Classia stats: 70 Damage, 40 AP Overclocked: 80 Damage, 60AP 15 shots with a cell 
Venatori stats: 70 Damage, 40AP Overclocked: 90 Damage, 60AP 15 shots with a cell 
Sollex stats: 70 Damage, 40AP Overclocked: 100 Damage, 60AP 10 shots with a cell 

Recoil is basically non-existent, if one is to simply take for the basic damage all the guns share, classia is the most efficient and powerful, because it does the same damage as the cannon without overclock, fires faster and can be dual wielded.

Cryo-sealed hydrogen fuel pack: 5.000 Units of plasma hydrogen, translating to 500 shots on the Classia and Venatori, 250 with the cannon. 

There is some strange quirks to the fuel pack, for one you can attach the guns to it normally, but that destroys the cell inside, which can be irritating, so you have to remove the cell prior to linking it to a pack, it also has a very strong tendency of creating a ghost item in hand, making your hand unusable, messing with them in one round ended up having me get ghost items 3 times in less than 1 hour. 

Continuing with the heat generation: The guns all have it, but the pack somehow disables that feature, with a classia linked to it, I was able to continuously spam the shooting, without getting heat enough to harm me, weird part is that I can dual wield classias, as I can remove em from the backpack, drop the backpack, walk away, as the code seems to try and recall the guns back, but fails at it, so one can link two classias, drop the packs, put classias in satchel, and walk around with quick firing explosives that don't generate heat and has 500 shots each.

- [ ] Sollex still has the malus of heat generation even when linked to the pack
- [ ] Classias when dual wielded DO end up causing heat generation somehow, but it still takes a lot more shots for it to happen, seems like the heat generation when dual wielded is shared between guns
- [ ] Venatori also has the same case as Classia when attached to the pack, as one can continuously spam shoot and not have any malus of heat generation

- [ ] One can also attach a cryo-sealed hydrogen fuel pack to the hydrogen grenade hilts (It doesn't make any bigger of an explosion, its just the same but with a bigger waste)
- [ ] The hydrogen generator, which uses em, can be made into a endless source of energy with no further material waste, the cells which it use, can be drained empty, but when recycled they will give the same amount of materials back, making the generator produce infinite energy as long the cells are recycled back
- [ ] The cells also return its full mat value when recycled, even when spent by gun usage, so in practice, hydrogen cells are a endless source of energy and ammo as long its recycled, printed, used and repeat
![image](https://github.com/user-attachments/assets/cd1bf149-1b43-4dbb-89dd-6c9ed8304cb2) 
Image shown: That being at T4, Using only 0.1 per tick

- [ ] Cryo-sealed hydrogen fuel pack can also be used by the hydrogen generator, giving it a source of 5000 fuel to use, unlike the limit of 150 which cells have, which is very likely enough to feed a single hydrogen generator for a full round at max potency if not more

**Comment by SigDoesCode** [comment]
Currently got a fix for the church sword spam.
Also working on a fix for sanity exploits. After that, going to fix the altar spam and fireman throwing.
